src/iris_api/utils.py

```diff
@@ -55,6 +55,16 @@ def parse_response(response, mode, source):
                                     LIMIT 1''', {'target_name': target_name}).scalar()
         session.close()
         return msg_id, 'claim'
+    elif re.match('[cC]laim\s+all', response):
+        session = db.Session()
+        target_name = lookup_username_from_contact(mode, source, session=session)
+        msg_ids = [row[0] for row in session.execute('''SELECT `message`.`id` from `message`
```

Comment:
Do this as part of atomic claiming in the DB instead of fetching a list then trying to claim each one in a CAS fashion to account for race conditions